x-pack/plugins/endpoint/server/routes/resolver/queries/base.ts

```diff
@@ -7,50 +7,91 @@
 import { SearchResponse } from 'elasticsearch';
 import { IScopedClusterClient } from 'kibana/server';
 import { ResolverEvent } from '../../../../common/types';
-import {
-  paginate,
-  paginatedResults,
-  PaginationParams,
-  PaginatedResults,
-} from '../utils/pagination';
 import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
 import { legacyEventIndexPattern } from './legacy_event_index_pattern';
+import { MSearchQuery } from './multi_searcher';
 
-export abstract class ResolverQuery {
-  constructor(
-    private readonly indexPattern: string,
-    private readonly endpointID?: string,
-    private readonly pagination?: PaginationParams
-  ) {}
-
-  protected paginateBy(tiebreaker: string, aggregator: string) {
-    return (query: JsonObject) => {
-      if (!this.pagination) {
-        return query;
-      }
-      return paginate(this.pagination, tiebreaker, aggregator, query);
-    };
+/**
+ * ResolverQuery provides the base structure for queries to retrieve events when building a resolver graph.
+ *
+ * @param T the structured return type of a resolver query. This represents the type that is returned when translating
+ * Elasticsearch's SearchResponse<ResolverEvent> response.
+ */
+export abstract class ResolverQuery<T> implements MSearchQuery {
+  constructor(private readonly indexPattern: string, private readonly endpointID?: string) {}
```

Comment:
Pagination is no longer required by this class. Each individual query defines wether it needs pagination or not.

Reply:
:grey_question: Is `endpointID` is only expected for legacy queries? If so a name like `legacyEndpointID` might reveal that a little better.